src/proto_reader.rs

```diff
@@ -12,26 +12,45 @@ pub struct ProtoReader {
     resource_reader: ResourceReader,
 }
 
+pub struct FileRead {
+    pub descriptor: FileDescriptorProto,
+    pub path: String,
+    pub content_ty: Option<String>,
+}
+
 pub struct ProtoMetadata {
     pub package: Option<String>,
     pub name: Option<String>,
```

Comment:
package and name is available in FileDescriptorSet.

Reply:
It is used to store package and name of the 1st file whose content we resolve.

It's used to set `src` for Link